### PR TITLE
c# Simple Types documentation

### PR DESCRIPTION
**Pull request by kevinchristianson**

### CODE REVIEW

File: Cogs.Design/CsSimpleType.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+built-in int type with Range and ExclusiveRange attributes
+
+## long
+built-in long type with Range and ExclusiveRange attributes
+
+## int
+built-in int type with Range and ExclusiveRange attributes
+
+## nonNegativeInteger
+built-in uint type with Range and ExclusiveRange attributes
```
Comment (DanSmith) [inline Cogs.Design/CsSimpleType.md]: Lets do a regular int for this

File: Cogs.Design/CsSimpleType.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+## unsignedLong
+built-in ulong type with Range and ExclusiveRange attributes
+
+## positiveInteger
```
Comment (DanSmith) [inline Cogs.Design/CsSimpleType.md]: Lets do a regular int for this

File: Cogs.Design/CsSimpleType.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+built-in uint type with Range and ExclusiveRange attributes
+
+## cogsDate
```
Comment (DanSmith) [inline Cogs.Design/CsSimpleType.md]: We can define this as a CogsDate type, which we can generate automatically along with the c# classes

File: Cogs.Model/Property.cs [hunk 4]
```diff
@@ -29,7 +29,16 @@ public bool IsPrimitive
...
         }
-
+        /*
+        public int? MinLength { get; set; } = null;
+        public int? MaxLength { get; set; } = null;
+        public int? MinInclusive { get; set; } = null;
+        public int? MinExclusive { get; set; } = null;
+        public int? MaxInclusive { get; set; } = null;
+        public int? MaxExclusive { get; set; } = null;
+        public Enum Enumerations { get; set; } = null;
```
Comment (DanSmith) [inline Cogs.Model/Property.cs]: We'll merge this, but I think we will have a List<string> for the accepted types. Enum would have to be known at compile time

File: Cogs.Publishers/CsSchemaPublisher.cs [hunk 5]
```diff
@@ -79,5 +104,32 @@ public void Publish(CogsModel model)
                 File.WriteAllText(Path.Combine(TargetDirectory, item.Name + ".cs"), newClass.ToString().Replace("#", "    ").Replace("$", Environment.NewLine));
             }
         }
+
+        // takes a data type name string and translates to a c# data structure representation name string
+        private string SetDataTypeName(string dataType)
+        {
+            if (dataType.Equals("boolean")) { return "bool"; }
```
Comment (DanSmith) [inline Cogs.Publishers/CsSchemaPublisher.cs]: change these string.Equals(dataType, "simpletype", StringComparison.OrdinalIgnoreCase)

File: Types.cs [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        }
+    }
+    */
+
+
+    public struct CogsDate
+    {
+        DateTimeOffset DateTime;
+        DateTimeOffset Date;
+        Tuple<int, int> GYearMonth;
```
Comment (DanSmith) [inline Types.cs]: Perhaps we could use a named tuple here? They are new in c# 7

File: Types.cs [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        int GYear;
+        TimeSpan Duration;
+        Type Type;
```
Comment (DanSmith) [inline Types.cs]: The Type would be the same for DateTime and Date, even though they are different properties. Perhaps we make an enumeration of the different available types, and set that.

public enum CogsDateType { ....... }

File: Types.cs (hunk 6 above)
Comment (kevinchristianson) [inline Types.cs]: In order to use named tuple we'd need to get the System.ValueTuple package from NuGet which isn't already installed (at least for me)

**Pull request merged**